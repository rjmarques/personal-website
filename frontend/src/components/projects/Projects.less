@import "../../theme.less";

.projects {
  .projects-action-icon {
    font-size: 24px;
  }

  .projects-card-row {
    display: grid;
    grid-column-gap: 16px;
    grid-row-gap: 16px;

    grid-template-columns: 1fr;

    .for-size(tablet-up; {
      grid-template-columns: repeat(2, 2fr);
    });
    .for-size(laptop-up; {
      grid-template-columns: repeat(3, 3fr);
    });
  }

  .projects-card {
    border-bottom: 4px solid @primary-color;
    cursor: auto;
    max-width: 100%;
    min-width: 0px;
    text-align: center;
    transition: all 0.3s;

    &:hover {
      border-bottom: 4px solid @primary-color;
      .for-size(laptop-small-up; {transform: translate(0px, -5px) ;});
    }

    a:hover,
    a:active,
    a:focus {
      color: @primary-color;
    }
  }

  .projects-card-subtitle {
    color: @grey;
    font-size: 13px;
    margin-top: 5px;
    white-space: nowrap;
  }
}
