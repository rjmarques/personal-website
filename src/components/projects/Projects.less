@import "../../theme.less";

@verticalGutter: 16px;

.Projects {
  .Projects-action-icon {
    font-size: 24px;
  }

  .Projects-card-row {
    margin-bottom: -@verticalGutter;
  }

  .Projects-card {
    border-bottom: 4px solid @primary-color;
    cursor: auto;
    margin-bottom: @verticalGutter;
    text-align: center;

    &:hover {
      border-bottom: 4px solid @primary-color;
      .for-size(laptop-small-up; {transform: translate(0px, -5px) ;});
    }

    a:hover,
    a:active,
    a:focus {
      color: @primary-color;
    }
  }

  .Projects-card-subtitle {
    color: @grey;
    font-size: 13px;
    margin-top: 5px;
    white-space: nowrap;
  }
}
