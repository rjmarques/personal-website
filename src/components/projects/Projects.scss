@use "../../theme.scss" as *;

.projects {
  .projects-card-row {
    display: grid;
    grid-column-gap: 16px;
    grid-row-gap: 16px;
    grid-template-columns: 1fr;

    @include for-size(tablet-up) {
      grid-template-columns: repeat(2, 2fr);
    }

    @include for-size(laptop-up) {
      grid-template-columns: repeat(3, 3fr);
    }
  }

  .projects-card {
    border: 1px solid $light-grey;
    border-bottom: 4px solid $primary-color;
    cursor: auto;
    max-width: 100%;
    min-width: 0px;
    text-align: center;
    transition: all 0.3s;

    &:hover {
      box-shadow: 0 5px 38px 2px rgba(0, 0, 0, 0.1);
      @include for-size(laptop-small-up) {
        transform: translate(0px, -5px);
      }
    }

    .projects-content {
      display: flex;
      flex-direction: column;
      align-items: center;
      padding: 24px 16px;

      h4 {
        text-align: center;
      }
    }

    .projects-actions {
      border-top: 1px solid $light-grey;
      display: flex;
      justify-content: center;
      margin: 0px;
      padding: 12px 0px
    }

    .projects-action-icon {
      background: none;
      flex: 1;    
      color: $grey;
      padding: 0px;
      transition: color 0.3s;

      &:hover {
        color: $primary-color;
      }

      svg {
        font-size: 26px;  
      }
    }

    a:hover,
    a:active,
    a:focus {
      color: $primary-color;
    }
  }

  .projects-card-subtitle {
    color: $grey;
    font-size: 13px;
    margin-top: 2px;
    white-space: nowrap;
  }
}
