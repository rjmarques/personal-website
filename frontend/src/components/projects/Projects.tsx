import { Card } from "antd";
import { PlayCircleOutlined, GithubOutlined } from "@ant-design/icons";
import React from "react";

import ContentSection from "../common/ContentSection";

import "./Projects.less";

interface IProject {
  description: string;
  image: any;
  repoUrl: string;
  title: string;
  url: string;
}

const projects: IProject[] = [
  {
    description: "A set of air quality sensors connected to an ESP32 MCU",
    image: require("./img/esp32.png"),
    repoUrl: "https://github.com/rjmarques/esp32-air-quality-sensor",
    title: "Esp32 Air Quality",
    url: "https://github.com/rjmarques/esp32-air-quality-sensor",
  },
  {
    description: "My own Terraform provisioned ECS cluster",
    image: require("./img/ecs.png"),
    repoUrl: "https://github.com/rjmarques/my-ecs-cluster",
    title: "My ECS Cluster",
    url: "https://github.com/rjmarques/my-ecs-cluster",
  },
  {
    description: "A twitter joke scraper running on ECS",
    image: require("./img/something-of-the-day.png"),
    repoUrl: "https://github.com/rjmarques/something-of-the-day",
    title: "Something of the day",
    url: "https://something.ricardomarques.dev/",
  },
  {
    description: "A Fractal WebAssembly Viewer",
    image: require("./img/wasm-mandel.png"),
    repoUrl: "https://github.com/rjmarques/webasm-mandelbrot",
    title: "WebAssembly Mandelbrot",
    url: "https://wasmmandel.ricardomarq.com/",
  },
  {
    description: "My Solar System",
    image: require("./img/solar-system.png"),
    repoUrl: "https://github.com/rjmarques/SolarSystem",
    title: "WebGL Solar System",
    url: "https://solar.ricardomarq.com/",
  },
];

const Projects = () => (
  <ContentSection
    title="Projects"
    subTitle="Some projects I've done in my free time"
    wrapperClass="projects"
  >
    <div className="projects-card-row materialize">
      {projects.map((proj) => (
        <Card
          key={proj.title}
          hoverable={true}
          className="projects-card"
          cover={<img alt="example" src={proj.image} />}
          actions={[
            <a
              key="demo_url"
              href={proj.url}
              target="_blank"
              rel="noopener noreferrer"
            >
              <PlayCircleOutlined className="projects-action-icon" />
            </a>,
            <a
              key="repo_url"
              href={proj.repoUrl}
              target="_blank"
              rel="noopener noreferrer"
            >
              <GithubOutlined className="projects-action-icon" />
            </a>,
          ]}
        >
          <h4>{proj.title}</h4>
          <span className="projects-card-subtitle">{proj.description}</span>
        </Card>
      ))}
    </div>
  </ContentSection>
);

export default Projects;
