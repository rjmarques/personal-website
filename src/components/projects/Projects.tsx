import { Card, Col, Icon, Row } from "antd";
import * as React from "react";

import SectionTitle from "../common/SectionTitle";

import "./Projects.less";

interface IProject {
  description: string;
  image: any;
  repoUrl: string;
  title: string;
  url: string;
}

class Projects extends React.Component {
  private projects: IProject[] = [
    {
      description: "An Angular webapp recipe for webprojects",
      image: require("./img/angular-typescript-recipe.png"),
      repoUrl: "https://github.com/rjmarques/angular2-typescript-recipe",
      title: "Angular 2 & Typescript Recipe",
      url: "https://angular2recipe.ricardomarq.com/"
    },
    {
      description: "A Fractal WebAssembly Viewer",
      image: require("./img/wasm-mandel.png"),
      repoUrl: "https://github.com/rjmarques/webasm-mandelbrot",
      title: "WebAssembly Mandelbrot",
      url: "https://wasmmandel.ricardomarq.com/"
    },
    {
      description: "My tiny solar system",
      image: require("./img/solar-system.png"),
      repoUrl: "https://github.com/rjmarques/SolarSystem",
      title: "WebGL Solar System",
      url: "https://solar.ricardomarq.com/"
    }
  ];

  public render() {
    return (
      <section className="Projects">
        <div className="content">
          <SectionTitle
            title="Projects"
            subTitle="Some projects I've done in my free time"
          />
          <div>{this.getProjects()}</div>
        </div>
      </section>
    );
  }

  private getProjects(): JSX.Element {
    return (
      <Row gutter={12}>
        {this.projects.map(proj => (
          <Col key={proj.title} xs={24} sm={24} md={8} lg={8} xl={8}>
            <Card
              hoverable={true}
              className="Projects-card"
              cover={<img alt="example" src={proj.image} />}
              actions={[
                <a key="demo_url" href={proj.url} target="_blank">
                  <Icon className="Projects-action-icon" type="play-circle" />
                </a>,
                <a key="repo_url" href={proj.repoUrl} target="_blank">
                  <Icon className="Projects-action-icon" type="github" />
                </a>
              ]}
            >
              <h4>{proj.title}</h4>
              <span className="Projects-card-subtitle">{proj.description}</span>
            </Card>
          </Col>
        ))}
      </Row>
    );
  }
}

export default Projects;
