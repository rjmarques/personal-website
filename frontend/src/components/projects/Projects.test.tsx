import React from "react";
import renderer from "react-test-renderer";

import Projects from "./Projects";

describe("renders without crashing", () => {
  it("renders correctly", async () => {
    const component = renderer.create(<Projects />);
    expect(component).toMatchSnapshot();
  });
});
