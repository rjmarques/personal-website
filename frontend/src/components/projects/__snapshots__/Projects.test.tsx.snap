// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders correctly 1`] = `
<section
  className="projects"
>
  <div
    className="content"
  >
    <div
      className="content-title"
    >
      <h2>
        Projects
      </h2>
      <span
        className="section-sub-title"
      >
        Some projects I've done in my free time
      </span>
    </div>
    <div
      className="projects-card-row materialize"
    >
      <div
        className="ant-card projects-card ant-card-bordered ant-card-hoverable"
      >
        <div
          className="ant-card-cover"
        >
          <img
            alt="example"
            src="esp32.png"
          />
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <h4>
            Esp32 Air Quality
          </h4>
          <span
            className="projects-card-subtitle"
          >
            A set of air quality sensors connected to an ESP32 MCU
          </span>
        </div>
        <ul
          className="ant-card-actions"
        >
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://github.com/rjmarques/esp32-air-quality-sensor"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="play-circle"
                  className="anticon anticon-play-circle projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="play-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    />
                    <path
                      d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://github.com/rjmarques/esp32-air-quality-sensor"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="github"
                  className="anticon anticon-github projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="github"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
        </ul>
      </div>
      <div
        className="ant-card projects-card ant-card-bordered ant-card-hoverable"
      >
        <div
          className="ant-card-cover"
        >
          <img
            alt="example"
            src="ecs.png"
          />
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <h4>
            My ECS Cluster
          </h4>
          <span
            className="projects-card-subtitle"
          >
            My own Terraform provisioned ECS cluster
          </span>
        </div>
        <ul
          className="ant-card-actions"
        >
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://github.com/rjmarques/my-ecs-cluster"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="play-circle"
                  className="anticon anticon-play-circle projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="play-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    />
                    <path
                      d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://github.com/rjmarques/my-ecs-cluster"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="github"
                  className="anticon anticon-github projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="github"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
        </ul>
      </div>
      <div
        className="ant-card projects-card ant-card-bordered ant-card-hoverable"
      >
        <div
          className="ant-card-cover"
        >
          <img
            alt="example"
            src="something-of-the-day.png"
          />
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <h4>
            Something of the day
          </h4>
          <span
            className="projects-card-subtitle"
          >
            A twitter joke scraper running on ECS
          </span>
        </div>
        <ul
          className="ant-card-actions"
        >
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://something.ricardomarques.dev/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="play-circle"
                  className="anticon anticon-play-circle projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="play-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    />
                    <path
                      d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://github.com/rjmarques/something-of-the-day"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="github"
                  className="anticon anticon-github projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="github"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
        </ul>
      </div>
      <div
        className="ant-card projects-card ant-card-bordered ant-card-hoverable"
      >
        <div
          className="ant-card-cover"
        >
          <img
            alt="example"
            src="wasm-mandel.png"
          />
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <h4>
            WebAssembly Mandelbrot
          </h4>
          <span
            className="projects-card-subtitle"
          >
            A Fractal WebAssembly Viewer
          </span>
        </div>
        <ul
          className="ant-card-actions"
        >
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://wasmmandel.ricardomarq.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="play-circle"
                  className="anticon anticon-play-circle projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="play-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    />
                    <path
                      d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://github.com/rjmarques/webasm-mandelbrot"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="github"
                  className="anticon anticon-github projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="github"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
        </ul>
      </div>
      <div
        className="ant-card projects-card ant-card-bordered ant-card-hoverable"
      >
        <div
          className="ant-card-cover"
        >
          <img
            alt="example"
            src="solar-system.png"
          />
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <h4>
            WebGL Solar System
          </h4>
          <span
            className="projects-card-subtitle"
          >
            My Solar System
          </span>
        </div>
        <ul
          className="ant-card-actions"
        >
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://solar.ricardomarq.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="play-circle"
                  className="anticon anticon-play-circle projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="play-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                    />
                    <path
                      d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
          <li
            style={
              Object {
                "width": "50%",
              }
            }
          >
            <span>
              <a
                href="https://github.com/rjmarques/SolarSystem"
                rel="noopener noreferrer"
                target="_blank"
              >
                <span
                  aria-label="github"
                  className="anticon anticon-github projects-action-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="github"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                    />
                  </svg>
                </span>
              </a>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
